File: python/j00.arguments.py
#!/usr/bin/python
import sys

# illustrative variables
ARG_COUNT   = len(sys.argv) - 1 # get num of real arguments
SCRIPT_NAME = sys.argv[0]       # get name of script

# check for only 2 arguments
if ARG_COUNT != 2:
   # ouptut usage statement to standard error
   sys.stderr.write("\nYou need to enter two numbers: \n\n")
   sys.stderr.write("   Usage: %s [num1] [num2]\n\n" % SCRIPT_NAME)
else:
   # get sum of both arguments, convert strings to integers
   sum = int(sys.argv[1]) + int(sys.argv[2])  
   # print results of both arguments (string) and summation (integer)
   print "The sum of %s and %s is: %d." % (sys.argv[1], sys.argv[2], sum)

File: php/j00.arguments.php
#!/usr/bin/env php
<?php
// acquire num of args and script name
$arg_count   = $argc - 1;  // get num of arguments
$script_name = $argv[0];   // get script name

if ($arg_count != 2) {
   # output usage statment to standard error
   fwrite(STDERR, "\nYou need to enter two numbers: \n\n");
   fwrite(STDERR, "   Usage: $script_name [num1] [num2]\n\n");
} else {
  $sum = $argv[1] + $argv[2]; // get sum of both arguments
  // print results of both arguments and summation
  echo  "The sum of $argv[1] and $argv[2] is: $sum.\n";
}
?>